src/SAML2/Protocol/Saml20LogoutHandler.cs

```diff
@@ -74,6 +74,10 @@ protected override void Handle(HttpContext context)
                 {
                     idpEndpoint = RetrieveIDPConfiguration(StateService.Get<string>(IdpLoginSessionKey));
                 }
+                else
+                {
+                    idpEndpoint = RetrieveIDP(context);
```

Comment:
Just a little uncertain as to why this is needed? A couple of things:
 - Couldn't RetrieveIDPConfiguration() be extended to contain the config from the file?
 - Is there a risk that the above function and the TransferClient() call below both do a redirect?